src/lib/factories.js

```diff
@@ -16,7 +16,7 @@ const mergePropsAndClassName = (defaultProps, props) => {
     newProps.className = cx(defaultProps.className, props.className) // eslint-disable-line react/prop-types
   }
 
-  if (!newProps.key) {
+  if (!newProps.key && childKey) {
```

Comment:
Prevent adding `childKey: undefined` due to props with no key and also not defining a `childKey`.  This issue was revealed by tests.